--- systrace-1.6e/linux_syscalls.c.orig	2007-12-19 05:59:30.000000000 +0100
+++ systrace-1.6e/linux_syscalls.c	2007-12-24 12:00:58.351116313 +0100
@@ -9,7 +9,7 @@
 /*
  * this file emulates openbsd's syscallnames[] array.  linux does not
  * have such a feature.  automatically generated from
- * arch/i386/kernel/entry.S
+ * arch/i386/kernel/syscall_table.S
  */
 
 char *linux_syscallnames[] = {
@@ -286,7 +286,7 @@
         "tgkill",        /* 270 */
         "utimes",
         "fadvise64_64",
-        "ni_syscall",    /* sys_vserver */
+        "vserver",    /* sys_vserver */
         "mbind",
         "get_mempolicy",
         "set_mempolicy",
@@ -302,17 +302,42 @@
         "add_key",
         "request_key",
         "keyctl",
-	"ni_syscall-31",
-	"ni_syscall-32",	 /* 290 */
-	"ni_syscall-33",
-	"ni_syscall-34",
-	"ni_syscall-35",
-	"ni_syscall-36",
-	"ni_syscall-37",	 /* 295 */
-	"ni_syscall-38",
-	"ni_syscall-39",
-	"ni_syscall-40",
-	"ni_syscall-41",
+	"ioprio_set",
+	"ioprio_get",		/* 290 */
+	"inotify_init",
+	"inotify_add_watch",
+	"inotify_rm_watch",
+	"migrate_pages",
+	"openat",		/* 295 */
+	"mkdirat",
+	"mknodat",
+	"fchownat",
+	"futimesat",
+	"fstatat64",		/* 300 */
+	"unlinkat",
+	"renameat",
+	"linkat",
+	"symlinkat",
+	"readlinkat",		/* 305 */
+	"fchmodat",
+	"faccessat",
+	"pselect6",
+	"ppoll",
+	"unshare",		/* 310 */
+	"set_robust_list",
+	"get_robust_list",
+	"splice",
+	"sync_file_range",
+	"tee",			/* 315 */
+	"vmsplice",
+	"move_pages",
+	"getcpu",
+	"epoll_pwait",
+	"utimensat",		/* 320 */
+	"signalfd",
+	"timerfd",
+	"eventfd",
+	"sys_fallocate",
 	NULL			 /* sentinel */
 };
 
